scripts/azure_cleanup.py

```diff
@@ -0,0 +1,44 @@
+import argparse
+import sys
+import traceback as tb
+from datetime import datetime
+
+from utils.path import log_path
+from utils.providers import list_provider_keys, get_mgmt
+
+
+def parse_cmd_line():
+    parser = argparse.ArgumentParser(argument_default=None)
+    parser.add_argument('--nic-template',
+                        help='NIC Name template to be removed', default="test*", type=str)
+    parser.add_argument('--pip-template',
+                        help='PIP Name template to be removed', default="test*", type=str)
+    parser.add_argument("--output", dest="output", help="target file name, default "
+                                                        "'cleanup_azure.log' in "
+                                                        "utils.path.log_path",
+                        default=log_path.join('cleanup_azure.log').strpath)
+    args = parser.parse_args()
+    return args
+
+
+def azure_cleanup(nic_template, pip_template, output):
+    with open(output, 'w') as report:
+        report.write('azure_cleanup.py, NICs and PIPs Cleanup')
+        report.write("\nDate: {}\n".format(datetime.now()))
+    try:
+        for provider_key in list_provider_keys('azure'):
+            provider_mgmt = get_mgmt(provider_key)
+            print ("----- Provider: {} -----".format(provider_key))
+            print ("Removing Nics with the name \"{}\"".format(nic_template))
```

Comment:
**Required**

This should be recorded in the report as well, ideally tabulated by provider key.

If you implement my request on wrapanapi#120, you'll have the list of resource names/ids/etc.